lib/rules/no-unused-properties.js

```diff
@@ -452,6 +454,35 @@ module.exports = {
           const propertyReferences =
             propertyReferenceExtractor.extractFromPattern(pattern)
           container.propertyReferencesForProps.push(propertyReferences)
+        },
+        onDefineModelEnter(node, model) {
+          if (!groups.has('props')) {
+            return
+          }
+          const container = getVueComponentPropertiesContainer(
+            context.getSourceCode().ast
+          )
+          if (
+            node.parent &&
+            node.parent.type === 'VariableDeclarator' &&
+            node.parent.init === node
+          ) {
+            // If store the return of defineModel() in a variable, we can mark the model prop as used if use that variable.
+            // If that variable is unused it will already be reported by `no-unused-var` rule.
```

Comment:
```suggestion
            // If the return value of defineModel() is stored in a variable, we can mark the model prop as used if that that variable is used.
            // If that variable is unused, it will already be reported by `no-unused-var` rule.
```